### ISSUE TITLE
PyLint reports "no assignment is done" but variable is changed in the scope of the procedure

### ISSUE THREAD

**Issue by motib79** [PyCQA/pylint#9522]
### Bug description

```python
I got the message:
Using global for 'ticker' but no assignment is done

for the code:
---
def remove_ticker_from_list(stock_index):
    global ticker
    ticker[stock_index]="NULL_STOCK"
```

### Command used

```shell
VSCode
```

### Pylint output

```python
mentioned in descr
```

### Expected behavior

do not report this error

### Pylint version

```shell
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)] on win32
```

### OS / Environment

Win10

**Comment by mbyrnepr2** [comment]
Thanks @motib79. 

`Using global for 'ticker' but no assignment is done` - in other words, the `global ticker` line is not necessary and can be removed.

Here is where `global` would be necessary:
```python
ticker = {1: 2}

def remove_ticker_from_list(stock_index):
    global ticker
    ticker = "NULL_STOCK"

remove_ticker_from_list('x')
print(ticker)

NULL_STOCK
```

Perhaps the [documentation](https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/global-variable-not-assigned.html) could be updated to make it more clear?

**Comment by motib79** [comment]
Thanks for the comments.
Ticker is a list. And modified in the scope of the procedure. Therefore I
believe global assignment is required as like in case of any other variable
changes. Looks like PyLint does not recognize list change as a change that
require global assignment .

**Comment by mbyrnepr2** [comment]
@motib79 I think that is the same situation for list or dict.
I think this is your example (expanded a bit to show the list definition outside the function).
Python is happy with this and `global` isn't needed here because you are not re-assigning `ticker` but rather some element contained within `ticker`. Global would be needed if `ticker` were a string, for example, because the assignment inside the function wouldn't affect `ticker` which is defined outside the function:

```python
(venv312) markbyrne@Marks-Air-2 programming % cat example.py 
ticker = [1, 2, 3]


def remove_ticker_from_list(stock_index):
    ticker[stock_index]="NULL_STOCK"

 
remove_ticker_from_list(0)
print(ticker)
(venv312) markbyrne@Marks-Air-2 programming % python example.py
['NULL_STOCK', 2, 3]

```

**Comment by motib79** [comment]
Ok, thanks for the clarification.
